### ISSUE TITLE
Moving RegressionTest.py forward.

### ISSUE THREAD

**Issue by mrshirts** [shirtsgroup/InterMol#17]
This is moving along!  Two things:
1. We should require that the .top and .gro files are named according to their directories, rather than having to list the hard coded files.  If they don't match (I've fixed most of them). then 'git mv XOLD XNEW' the file so it is named correctly.
2. We want to move to a philosophy of rather than having to wrote n^2 separate sections, having loops over the sections. Thus you could have N sections describing the N files format in, and then each calls the same output file function, which takes an argument specified at the beginning as to which output programs you are testing.

Better yet would have N reading functions defined, then do a double loop over N reading functions and N output functions.

Basically, we want to avoid anything with hard coding things like 'GromacsToGromacs' and instead be looping over sets of programs and writing things like outputstr = program1 + 'To' + program2
